### PR TITLE
Fix improper Input usage in Gallery's Expansion demo.

### LINKED ISSUE #6011: Cursor handle isn't under cursor in text field
Open Gallery, go into Expansion Panels, tap into Trip name, then tap into the Trip name text field.

Notice how the cursor is at the right of the string, but the cursor handle is at the left of the string.

(The expectation is that the cursor handle is under the cursor.)

### PR DESCRIPTION
**Pull request by mpcomplete**
Fixes https://github.com/flutter/flutter/issues/6011

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 1]
```diff
@@ -191,17 +191,15 @@ class _ExpansionPanelsDemoState extends State<ExpasionPanelsDemo> {
...
-            child: new Input(
-              hintText: item.hint,
-              labelText: item.name,
-              value: new InputValue(
-                text: item.value
+            child: new Form(
+              child: new Input(
+                hintText: item.hint,
+                labelText: item.name,
+                value: new InputValue(text: item.value),
```
Comment (abarth) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: I don't think this is correct.  Creating a new InputValue each time will lose information about the selection and editing state.  Is there a need to use a Form here?  Why not just store the InputValue as state and return the InputValue you get in the onChanged callback?

Comment (mpcomplete) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: I think this is an indication that the API is confusing. But the way it works is that when there's a formField, the `value` param is only used for the initial value. Subsequent values are taken from the Form.

What you suggest should work, however. I just think using Form looks simpler/cleaner.

**Review by dragostis — APPROVED** [review]
LGTM. Thanks a lot, @mpcomplete! 😃

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart (hunk 1 above)
Comment (abarth) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: Yeah, there's a bug assigned to me to make this less confusing.

**Pull request merged**